cmake/Modules/FindTinyXML2.cmake

```diff
@@ -13,7 +13,26 @@ if(TinyXML2_FOUND)
       set(TINYXML2_LIBRARY tinyxml2)
     elseif(TARGET tinyxml2::tinyxml2)
       set(TINYXML2_LIBRARY tinyxml2::tinyxml2)
+    elseif(TinyXML2_FIND_REQUIRED)
+      message(FATAL_WARNING "Unable to determine target for TinyXML2")
     endif()
+    list(APPEND TinyXML2_TARGETS ${TINYXML2_LIBRARY})
+  else()
+    # Only perform that logic once
+    if(NOT TARGET tinyxml2::tinyxml2)
+      add_library(tinyxml2::tinyxml2 UNKNOWN IMPORTED)
+      # TINYXML2_LIBRARY is composed of debug;<path\to\debug.lib>;optimized;<path\to\release.lib>
+      # we have to extract the appropriate component based on the current configuration.
```

Comment:
Please add assert to ensure that assumption is actually true otherwise this might break silently in the future.

Reply:
True. I am not too happy about this change myself. Do you know of any good way to assert this here? I could check that entry `0` and `3` indeed have `debug` and `optimized` in it and maybe there is a way to validate that the file paths in `1` and `2` are valid.
Is there anything smarter to do in cmake?

Reply:
> Do you know of any good way to assert this here?

See the existing function with `assert` in the name in `ament_cmake_core` for an example.

> I could check that entry `0` and `3` indeed have `debug` and `optimized` in it

That sounds good - 0 and 2 though I guess.

Reply:
The latest commits should hopefully address this. I passes on my local Windows VM (I'd assume it will pass on CI as well), but I have no idea if there are other variants of this where the list might maybe only contain 2 items (either debug or optimized for example).